src/bp-templates/bp-nouveau/includes/messages/ajax.php

```diff
@@ -1082,7 +1082,7 @@ function bp_nouveau_ajax_get_user_message_threads() {
 			'group_message_fresh'             => $group_message_fresh,
 			'excerpt'                         => wp_strip_all_tags( bp_get_message_thread_excerpt() ),
 			'content'                         => do_shortcode( bp_get_message_thread_content() ),
-			'unread'                          => bp_message_thread_has_unread(),
+			'unread'                          => bb_get_messages_thread_unread_count( $bp_get_message_thread_id ) ? true : false,
```

Comment:
```suggestion
			'unread'                          => bp_message_thread_has_unread(),
```
@krunal-buddyboss We don't want to change anything here revert this change.

Reply:
Done, Applied changes.